src/view/dataElements/xdmObject.jsx

```diff
@@ -248,24 +248,36 @@ const XdmExtensionView = () => {
           imsAccess: initInfo.tokens.imsAccess
         })
           .then(response => {
-            let sandbox = { name: null };
+            let sandbox;
             // if sandboxes are returned from the api call
             if (response.sandboxes && response.sandboxes.length) {
               setSandboxes(response);
-              // default to the first item in the list
-              sandbox = response.sandboxes.find(
-                _sandbox => _sandbox.isDefault === true
-              );
 
               // if a sandbox exists in settings and is in the list, choose that item
               if (initInfo.settings && initInfo.settings.sandbox) {
                 sandbox = response.sandboxes.find(
                   _sandbox => _sandbox.name === initInfo.settings.sandbox.name
                 );
               }
-              setSelectedSandbox(sandbox);
+
+              // default to the first item in the list
+              if (!sandbox) {
+                sandbox = response.sandboxes.find(
+                  _sandbox => _sandbox.isDefault
+                );
+              }
+
+              if (!sandbox) {
+                [sandbox] = response.sandboxes;
+              }
             }
 
+            if (!sandbox) {
+              sandbox = { name: null };
```

Comment:
I moved this from a different spot. I don't like this idea of creating a half-baked sandbox object and I'm not sure it's actually necessary. Some observations: 

1. `fetchSandboxes` ensures that if the response body has a `sandboxes` property that's an empty array, that it automatically adds a default sandbox object: https://github.com/adobe/reactor-extension-alloy/blob/c58174792eb61226183fef8f6c3a8d2c1c56202a/src/view/dataElements/xdmObject/helpers/fetchSandboxes.js#L47-L49

2. The code above this line wrapped in `if (response.sandboxes && response.sandboxes.length) {` ensures that if there's at least one sandbox in `response.sandboxes`, that the `sandbox` variable _will_ get set.

Based on those observations, it seems that we would only be creating this half-baked sandbox object if we get a response and the response didn't have a `sandboxes` property on it. I'm not sure when that would occur and I don't see anything in the API docs suggesting that might occur. I also don't see any functional test for it. It might just be dead code. @garyr maybe you can provide some insight here?

Reply:
I plan on removing the half-baked sandbox object in the react-spectrum upgrade btw.

Reply:
Sounds like a reasonable explanation. I've added a commit to remove this partial sandbox object. If we do find there's a case for it, we'll solve the issue and add a functional test for it.